keps/sig-network/4962-network-topology-standard/README.md

```diff
@@ -0,0 +1,867 @@
+2. `block`: Rack-level switches connecting hosts in one or more racks as a block.
+3. `spine`: Spine-level switches connecting multiple blocks inside a datacenter.
+4. `datacenter`: Zonal switches connecting multiple datacenters inside an availability zone.
+
+These types will accommodate the majority of common network hierarchies across different CSP and on-prem environments.
+Having these labels available in Kubernetes clusters will help in designing cloud agnostic scheduling systems.
+The scheduler will prioritize hierarchy layers according to the order outlined above, providing a standardized
+approach for network-aware scheduling across a range of configurations.
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a data scientist running a data-intensive large-scale AI training job, I want to optimize the runtime
+by binding pods to nodes that are in close network proximity.
+This ensures better performance for my distributed workloads.
+
+Additionally, I do not want to modify the job specification when migrating the job across different Kubernetes environments.
+
+To achieve this, I can leverage the pod affinity feature of the default Kubernetes scheduler with topology keys:
+```yaml
+  spec:
+    affinity:
+      podAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 70
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                  - key: app
+                    operator: In
+                    values:
+                      - training
+              topologyKey: network.topology.kubernetes.io/block
+          - weight: 90
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                  - key: app
+                    operator: In
+                    values:
+                      - training
+              topologyKey: network.topology.kubernetes.io/accelerator
+```
+#### Story 2
+
+As a developer, I would like to extend Kubernetes-native scheduling capabilities to
+support gang-scheduling for multi-node jobs.
+
+My goal is to ensure that this plugin remains cloud-agnostic. The design leverages the
+presence of `network.topology.kubernetes.io/...` node labels to reconstruct the cluster
+network topology and implement a network-aware placement algorithm.
+
+### Notes/Constraints/Caveats (Optional)
+
+The delivery method for the cluster network topology lies outside the scope of this proposal.
+However, this information could be:
+- Provided directly by CSPs, where CSPs apply node labels during node creation.
+- Extracted from CSPs using specialized tools like [Topograph](https://github.com/NVIDIA/topograph).
```

Comment:
This would mean users need to modify the job spec moving across different Kubernetes environments right?

Reply:
If the community adopts this KEP, users won't need to modify the job spec when changing K8s environments.